scripts/kind-e2e/lib_subctl_gather_test.sh

```diff
@@ -32,13 +32,27 @@ function test_subctl_gather() {
   validate_resource_files $subm_ns 'deployments' 'Deployment' '-l app=submariner-networkplugin-syncer'
   validate_resource_files $subm_ns 'deployments' 'Deployment' '-l app=submariner-lighthouse-agent'
   validate_resource_files $subm_ns 'deployments' 'Deployment' '-l app=submariner-lighthouse-coredns'
+
+  # Service Discovery
+  validate_resource_files all 'serviceexports.multicluster.x-k8s.io' 'ServiceExport'
+  validate_resource_files all 'serviceimports.multicluster.x-k8s.io' 'ServiceImport'
+  validate_resource_files all 'endpointslices.discovery.k8s.io' 'EndpointSlice'
+  validate_resource_files $subm_ns 'configmaps' 'ConfigMap' '-l component=submariner-lighthouse'
+  validate_resource_files kube-system 'configmaps' 'ConfigMap' '-l k8s-app=kube-dns'
```

Comment:
Looking at the e2e test [output](https://github.com/submariner-io/submariner-operator/pull/1196/checks?check_run_id=2243724486), it generated the configmap file:

`11781 ✓ Found 1 configmaps by field selector "metadata.name=coredns" in namespace "kube-system"`

but this command to retrieve it didn't find it so it wasn't verified:

`13270 [submariner-operator]$ [cluster2] kubectl --context=cluster2 get configmaps --namespace=kube-system -l k8s-app=kube-dns -o=jsonpath={.items..metadata.name}`

We need to use a `metadata.name` field selector for lookup:

```suggestion
  validate_resource_files kube-system 'configmaps' 'ConfigMap' '--field-selector metadata.name=coredns'
```